Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -1476,10 +1483,68 @@ public void RegisterIndicator<T>(Symbol symbol, IndicatorBase<T> indicator, IDat
             };
         }
 
+        /// <summary>
+        /// Warms up a given indicator with historical data
+        /// </summary>
+        /// <param name="symbol">The symbol whose indicator we want</param>
+        /// <param name="indicator">The indicator we want to warm up</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData to send into the indicator, if null defaults to the Value property of BaseData (x => x.Value)</param>
+        /// <returns>The given indicator</returns>
+        public IndicatorBase<IndicatorDataPoint> WarmUpIndicator(Symbol symbol, IndicatorBase<IndicatorDataPoint> indicator, Resolution? resolution = null, Func<IBaseData, decimal> selector = null)
+        {
+            // assign default using cast
+            selector = selector ?? (x => x.Value);
+
+            GetIndicatorWarmUpHistory(symbol, indicator, resolution)
+                .PushThrough(bar =>
+                {
+                    var input = new IndicatorDataPoint(bar.Symbol, bar.EndTime, selector(bar));
+                    indicator.Update(input);
+                });
+
+            return indicator;
+        }
+
+        /// <summary>
+        /// Warms up a given indicator with historical data
+        /// </summary>
+        /// <param name="symbol">The symbol whose indicator we want</param>
+        /// <param name="indicator">The indicator we want to warm up</param>
+        /// <param name="resolution">The resolution</param>
+        /// <param name="selector">Selects a value from the BaseData send into the indicator, if null defaults to a cast (x => (T)x)</param>
+        /// <returns>The given indicator</returns>
+        public IndicatorBase<T> WarmUpIndicator<T>(Symbol symbol, IndicatorBase<T> indicator, Resolution? resolution = null, Func<IBaseData, T> selector = null)
+            where T : IBaseData
+        {
+            // assign default using cast
+            selector = selector ?? (x => (T)x);
+
+            GetIndicatorWarmUpHistory(symbol, indicator, resolution)
```

Comment:
This method should work regardless of the boolean setting to enable manual usage. The boolean should be used in the Indicator helper methods (SMA, EMA) to warm them up before returning.